lib/mobility/arel/nodes/pg_ops.rb

```diff
@@ -29,54 +29,66 @@ def lower
         end)
       end
 
-      JsonbDashArrow.class_eval do
-        def quoted_node(other)
-          other && super(other.to_json)
-        end
-      end
-
       # Needed for AR 4.2, can be removed when support is deprecated
       HstoreDashArrow.class_eval do
         def quoted_node(other)
           other && super
         end
       end
 
-      class Jsonb < JsonbDashArrow
-        def matches *args
-          JsonbDashDoubleArrow.new(left, right).matches(*args)
+      class Jsonb  < JsonbDashDoubleArrow
+        def to_dash_arrow
+          JsonbDashArrow.new(left, right)
         end
 
-        def lower
-          JsonDashDoubleArrow.new(left, right).lower
+        def to_question
+          JsonbQuestion.new(left, right)
+        end
+
+        def eq other
+          case other
+          when NilClass
+            to_question.not
+          when Integer, Array, Hash
+            to_dash_arrow.eq(other.to_json)
```

Comment:
Previously Jsonb used the default `->` operator, but this is problematic in general, so instead we now use `->>` which returns a string and does not require casting. In case we are calling `eq` with an argument that is an integer, array or hash, we convert the argument to json and compare using `->` as before.

Non-string translations are an edge case, but one that I want to support.